src/V3DfgBreakCycles.cpp

```diff
@@ -0,0 +1,488 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Converting cyclic DFGs into acyclic DFGs
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2025 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+
+#include "V3PchAstNoMT.h"  // VL_MT_DISABLED_CODE_UNIT
+
+#include "V3Dfg.h"
+#include "V3DfgPasses.h"
+#include "V3Hash.h"
+
+#include <fstream>
+#include <limits>
+#include <unordered_map>
+#include <vector>
+
+VL_DEFINE_DEBUG_FUNCTIONS;
+
+// Similar algorithm used in ExtractCyclicComponents.
```

Comment:
Perhaps should put both graph algorithms into a  DfgGraphAlg header?

Reply:
I will try to unify ExtractCyclicComponents with the one in the cyclic decomposition one in a later patch.